components/tls/tls-mpc/src/leader.rs

```diff
@@ -639,13 +673,52 @@ impl Backend for MpcTlsLeader {
     }
 
     async fn buffer_incoming(&mut self, msg: OpaqueMessage) -> Result<(), BackendError> {
-        self.committed.push_back(msg);
+        if self.committed {
+            return Err(BackendError::InternalError(
+                "cannot buffer messages after committing to transcript".to_string(),
+            ));
+        }
+
+        if msg.typ == ContentType::ApplicationData {
+            self.channel
+                .send(MpcTlsMessage::CommitMessage(CommitMessage {
+                    msg: msg.payload.0.clone(),
+                }))
+                .await
+                .map_err(|e| BackendError::InternalError(e.to_string()))?;
+        }
+
+        self.buffer.push_back(msg);
+
+        if self.is_decrypting {
```

Comment:
Note to self: same q here as above about what the purpose of `is_decrypting` is.